app/models/course/assessment/assessment_ability.rb

```diff
@@ -48,7 +48,7 @@ def allow_read_assessments
   # 'access' refers to the ability to access password-protected assessments.
   def allow_access_assessment
     can :access, Course::Assessment do |assessment|
-      if assessment.view_password_protected?
+      if assessment.view_password_protected? && !assessment.is_koditsu_enabled
```

Comment:
I prefer you separate the logic:
```rb
if assessment.is_koditsu_enabled
  true # leave comment explaining this
else if ...
```
so it's clear to future developer why there is a bypass.

Reply:
done